src/storage/patch/RdfPatcher.ts

```diff
@@ -0,0 +1,70 @@
+import type { RdfDatasetRepresentation } from '../../http/representation/RdfDatasetRepresentation';
+import type { Representation } from '../../http/representation/Representation';
+import { RepresentationMetadata } from '../../http/representation/RepresentationMetadata';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { INTERNAL_QUADS } from '../../util/ContentTypes';
+import { InternalServerError } from '../../util/errors/InternalServerError';
+import { readableToQuads } from '../../util/StreamUtil';
+import type { RepresentationPatcherInput } from './RepresentationPatcher';
+import { RepresentationPatcher } from './RepresentationPatcher';
+
+/**
+ * Patcher that converts the representation to an N3 Store, does the patch using this store
+ * and then converts the store back to a representation which gets returned
+ */
+export class RdfPatcher extends RepresentationPatcher<Representation> {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly patcher: RepresentationPatcher<RdfDatasetRepresentation>;
+
+  public constructor(patcher: RepresentationPatcher<RdfDatasetRepresentation>) {
+    super();
+    this.patcher = patcher;
+  }
+
+  public async canHandle({ identifier, patch, representation }: RepresentationPatcherInput<Representation>):
+  Promise<void> {
+    // Wat als representation undefined is?
+    const newRepresentation: RdfDatasetRepresentation = representation as RdfDatasetRepresentation;
+    if (representation) {
+      newRepresentation.dataset = new Store();
+    }
+    await this.patcher.canHandle({ identifier, patch, representation: newRepresentation });
+  }
+
+  public async handle({ identifier, patch, representation }: RepresentationPatcherInput<Representation>):
+  Promise<Representation> {
+    let metadata = new RepresentationMetadata(identifier, INTERNAL_QUADS);
+    if (representation && representation.metadata.contentType !== INTERNAL_QUADS) {
+      this.logger.error('Received non-quad data. This should not happen so there is probably a configuration error.');
+      throw new InternalServerError('Quad stream was expected for patching.');
+    }
+
+    // Drain representation data to N3 Store
+    const inputRepresentation: RdfDatasetRepresentation = representation ?
+      representation as RdfDatasetRepresentation :
+      new BasicRepresentation() as RdfDatasetRepresentation;
+
+    if (representation) {
+      inputRepresentation.dataset = await readableToQuads(representation.data);
+      // eslint-disable-next-line prefer-destructuring
+      metadata = representation.metadata;
+    } else {
+      inputRepresentation.dataset = new Store();
+    }
+
+    // Execute the patcher
+    const patchedRepresentation = await this.patcher.handle({
+      identifier,
+      patch,
+      representation: inputRepresentation,
+    });
+
+    // Return the n3 store to the representation
+    const data = patchedRepresentation.dataset.match() as unknown as Readable;
```

Comment:
Perhaps add a comment saying we do this cast due to the `N3.js` typings not being precise enough, so we might update this in the future.